app/models/spree/image.rb

```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require "mini_magick"
```

Comment:
The alternative is to rescue `StandardError` which may be more robust and agnostic of what kind of problem is arising when processing an image.

Usually it's good practice to rescue specific errors. But if an unknown error breaks the shop page instead of not displaying an image then that's a bad consequence in production as well.

I want to merge this pull request today to include it in the release. But let's discuss other options, @openfoodfoundation/developers.

Reply:
I agree that using specific errors for rescuing should be the practice, however, I think it should be the case where we want to handle different errors differently.

If we need to handle each error similarly, then the `StandardError` rescue seems better with some logging or non-code-breaking alerts. In this case, we are rescuing from [3 different errors](https://github.com/mkllnk/openfoodnetwork/blob/93edf4e3adbc21580713a46fd0e65ebb37beb2e8/app/models/spree/image.rb#L34-L43) I guess, but each is handled in a same way. We could shorten this by just rescuing `StandardError`. In image processing, many cases may break the code and we may again face a similar issue in the future. That may have been handled by the `MiniMagick::Error` rescue. But yes, like you said `StandardError` may be more agnostic to handling these image processing cases.

Reply:
I agree it's worth breaking our rule of "always rescue specific errors" for this case. We are still notifying to Bugsnag so it won't be hiding any new types of errors.

Do we need to differentiate between known errors (ActiveStorage::Error, MiniMagick::Error, ActionView::Template::Error) and new error types? 
I think not. Whatever the error type, if it happens infrequently  we can probably ignore it. If it is impacting on users and they report it, we can investigate. Or if we see a huge spike in bugsnag, we can investigate it.

So I'm happy to switch to just StandardError. But I would suggest we use at least one of these known error types in the spec (if not already).